ovn_k8s/modes/overlay.py

```diff
@@ -304,3 +312,124 @@ def add_endpoint(self, event):
             return
 
         self._update_vip(service_data, ips)
+
+    def sync_pods(self, pods, namespace):
+        expected_logical_ports = []
+        if pods and 'items' in pods:
+            for pod in pods['items']:
+                pod_name = pod['metadata']['name']
+                logical_port = "%s_%s" % (namespace, pod_name)
+                expected_logical_ports.append(logical_port)
+
+        try:
+            existing_logical_ports = ovn_nbctl(
+                                "--data=bare", "--no-heading",
+                                "--columns=name", "find",
+                                "logical_switch_port",
+                                "external_ids:namespace=" + namespace).split()
+        except Exception as e:
+            vlog.err("sync_pods: find failed %s" % (str(e)))
+
+        for logical_port in existing_logical_ports:
+            if logical_port not in expected_logical_ports:
```

Comment:
If you build both expected_logical_ports and existing_logical_ports as sets rather than lists (it's ok as we won't have dupes), then here you can use set logic and do the following:

```
for logical_port in existing_logical_ports - expected_logical_ports:
   ....
```
This could reduce complexity from n^2 to n (I hope I'm not wrong here ;) )